orchestrator/careplancontributor/service.go

```diff
@@ -339,6 +337,66 @@ func (s Service) handleProxyAppRequestToEHR(writer http.ResponseWriter, request
 	}
 }
 
+func (s Service) handleSubscribeToTask(writer http.ResponseWriter, request *http.Request, session *user.SessionData) {
+	launchedTaskIdentifier := session.StringValues["taskIdentifier"]
+
+	if launchedTaskIdentifier == "" {
+		coolfhir.WriteOperationOutcomeFromError(request.Context(), coolfhir.BadRequest("No taskIdentifier found in session"), fmt.Sprintf("CarePlanContributor/%s %s", request.Method, request.URL.Path), writer)
+		return
+	}
+
+	sessionTaskIdentifier, err := coolfhir.TokenToIdentifier(launchedTaskIdentifier)
+	if err != nil {
+		coolfhir.WriteOperationOutcomeFromError(request.Context(), coolfhir.BadRequest(fmt.Sprintf("Invalid taskIdentifier in session: %v", err)), fmt.Sprintf("CarePlanContributor/%s %s", request.Method, request.URL.Path), writer)
+		return
+	}
+
+	if s.localCarePlanServiceUrl == nil {
+		coolfhir.WriteOperationOutcomeFromError(request.Context(), coolfhir.BadRequest("No local CarePlanService configured - cannot verify Task identifiers"), fmt.Sprintf("CarePlanContributor/%s %s", request.Method, request.URL.Path), writer)
```

Comment:
I think this isn't possible, because the frontend will then break as well. But fine to leave in.

Reply:
Very true! Can't hurt indeed. In case that logic changes, and this part is forgotten, it gives a clear message of what's going wrong